cmd/sysdump.go

```diff
@@ -0,0 +1,183 @@
+/*
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Copyright (c) 2022-2024, daeuniverse Organization <dae@v2raya.org>
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"io/ioutil"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+    
+	"github.com/vishvananda/netlink"
+	"github.com/spf13/cobra"
+	"github.com/mholt/archiver/v3"
+	"github.com/shirou/gopsutil/v4/net"
+)
+
+var (
+	sysdumpCmd = &cobra.Command{
+		Use:   "sysdump",
+		Short: "To dump up system network config",
+		Run: func(cmd *cobra.Command, args []string) {
+			dumpNetworkInfo()
+		},
+	}
+)
+
+func dumpNetworkInfo() {
+	tempDir, err := ioutil.TempDir("", "sysdump")
+	if err != nil {
+		fmt.Printf("Failed to create temp directory: %v\n", err)
+		return
+	}
+	defer os.RemoveAll(tempDir)
+
+	dumpRouting(tempDir)
+	dumpNetInterfaces(tempDir)
+	dumpSysctl(tempDir)
+	dumpNetfilter(tempDir)
+	dumpIPTables(tempDir)
+
+	tarFile := "sysdump.tar.gz"
+	if err := archiver.Archive([]string{tempDir}, tarFile); err != nil {
+		fmt.Printf("Failed to create tar archive: %v\n", err)
+		return
+	}
+
+	fmt.Printf("System network information collected and saved to %s\n", tarFile)
+}
+
+func dumpRouting(outputDir string) {
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		fmt.Printf("Failed to get routing table: %v\n", err)
+		return
+	}
+
+	var buffer bytes.Buffer
+	buffer.WriteString("Routing Table:\n")
+	for _, route := range routes {
+		link, err := netlink.LinkByIndex(route.LinkIndex)
+		if err != nil {
+			fmt.Printf("Failed to get link by index: %v\n", err)
+			continue
+		}
+		ifaceName := link.Attrs().Name
+
+		routeStr := ""
+		if route.Dst == nil {
+			routeStr += "default"
+		} else {
+			routeStr += route.Dst.String()
+		}
+
+		if route.Gw != nil {
+			routeStr += fmt.Sprintf(" via %s", route.Gw.String())
+		}
+
+		routeStr += fmt.Sprintf(" dev %s", ifaceName)
+
+		if route.Scope != 0 {
+			routeStr += fmt.Sprintf(" scope %d", route.Scope)
```

Comment:
Would you like to translate scope enum into semantic words? You can use constants defined in unix package: https://pkg.go.dev/github.com/vishvananda/netlink#Scope 
```
const (
	SCOPE_UNIVERSE Scope = unix.RT_SCOPE_UNIVERSE
	SCOPE_SITE     Scope = unix.RT_SCOPE_SITE
	SCOPE_LINK     Scope = unix.RT_SCOPE_LINK
	SCOPE_HOST     Scope = unix.RT_SCOPE_HOST
	SCOPE_NOWHERE  Scope = unix.RT_SCOPE_NOWHERE
)
```